_posts/2025-07-07-building-a-multi-modal-search-engine-in-opensearch.md

```diff
@@ -0,0 +1,309 @@
+---
+layout: post
+title:  "Building a multimodal search engine in OpenSearch"
+authors:
+ - mingshl
+ - kolchfa
+date: 2025-07-07
+categories:
+ - technical-post
+meta_keywords: "multimodal search, opensearch, semantic search, vector search, image search"
+meta_description: "Learn how to build a multimodal search engine in OpenSearch using the ML inference ingest and search request processors to search across text and images."
```

Comment:
Update meta: 
_meta_keywords:_ multimodal search, multimodality, semantic search, vector embeddings, machine learning, multimodal search in OpenSearch, deep neural networks, multimodal embedding, Titan Multimodal Embeddings, ML inference processor, k-NN vector

_meta_description:_ Learn how to build a powerful multimodal search engine in OpenSearch that processes text, images, and other data types simultaneously using ML inference processors and Amazon Titan models.